source/common/stats/stats_matcher_impl.cc

```diff
@@ -0,0 +1,44 @@
+#include "common/stats/stats_matcher_impl.h"
+
+#include <regex>
+#include <string>
+
+#include "common/common/utility.h"
+
+namespace Envoy {
+namespace Stats {
+
+StatsMatcherImpl::StatsMatcherImpl(const envoy::config::metrics::v2::StatsConfig& config) {
+  switch (config.stats_matcher().stats_matcher_case()) {
+  case envoy::config::metrics::v2::StatsMatcher::kExclusionList:
+    for (const auto& stats_matcher : config.stats_matcher().exclusion_list().patterns()) {
+      matchers_.push_back(Matchers::StringMatcher(stats_matcher));
+    }
+    break;
+  case envoy::config::metrics::v2::StatsMatcher::kInclusionList:
+    default_inclusive_ = false;
+    for (const auto& stats_matcher : config.stats_matcher().inclusion_list().patterns()) {
+      matchers_.push_back(Matchers::StringMatcher(stats_matcher));
+    }
+    break;
+  default:
+    // No matcher was supplied, so we default to allow all stat names.
+    break;
+  }
+}
+
+bool StatsMatcherImpl::rejects(const std::string& name) const {
+  if (default_inclusive_) {
```

Comment:
I think you could get rid of the branch here. You could do something like:
```
return default_inclusive_ == std::any_of(matchers_.begin(), matchers_.end(),
                       [&name](auto matcher) { return matcher.match(name); });
```

That's certainly harder to read than what you have, but I'm generally of the opinion that if you can save a branch it's worth forcing the reader to read an explanatory comment, but it's ultimately up to you.

If you do want to keep the branching for readability, I'd suggest something like this to avoid repeating the same logic:
```
bool did_match = std::any_of(matchers_.begin(), matchers_.end(),
                       [&name](auto matcher) { return matcher.match(name); });
if (default_inclusive_) {
    return did_match;
} else {
    return !did_match;
}
```

Reply:
This is a great catch -- I think you're right, `==` will be faster once optimized. I've written it that way but left a truth table explaining why XNOR is the correct implementation of this branching model. See 51b84eb.